src/Extensions/ElementalAreasExtension.php

```diff
@@ -223,24 +223,32 @@ protected function onBeforeWrite()
         if (!$this->supportsElemental()) {
             return;
         }
-
-        $elementalAreaRelations = $this->owner->getElementalRelations();
-
+        $owner = $this->getOwner();
+        $elementalAreaRelations = $owner->getElementalRelations();
         $this->ensureElementalAreasExist($elementalAreaRelations);
-
-        $ownerClassName = get_class($this->owner);
+        $ownerClassName = $owner->ClassName;
 
         // Update the OwnerClassName on EA if the class has changed
-        foreach ($elementalAreaRelations as $eaRelation) {
-            $ea = $this->owner->$eaRelation();
-            if ($ea->OwnerClassName !== $ownerClassName) {
-                $ea->OwnerClassName = $ownerClassName;
-                $ea->write();
+        $doUpdate = true;
+        // Do not update if using live versioning mode i.e. publishing,
+        // as it can result in a duplicate ElementalArea being created
+        if ($owner::has_extension(Versioned::class)
+            && Versioned::get_reading_mode() === 'Stage.' . Versioned::LIVE
```

Comment:
@GuySartorelli You mentioned you were going to have have a look during peer review to see if there was a better way to do this check